### ISSUE TITLE
Fix TerminationGracePeriodSeconds is negative 

### ISSUE THREAD

**Issue by wzshiming** [kubernetes/kubernetes#103476]
* [x] Firstly release 1.22 #98866: 
     * Treat negative as 1 in delete path (pod/strategy.go, apiserver delete.go)
     * Allow delete with grace period 0 even if metadata period is negative (apiserver delete.go). This allows kubelet to delete pods.
     * Relax validation of immutable fields to allow it to be set to 1 if it was previously negative. This allows us to update read path in the next release. All consumers would also need to be able to handle this change in a sane manner.
 * [x] Secondly release 1.27 #115606:
     * Treat negative as 1 (defaulting or conversion)
 * [ ] Thirdly:
     * Returning an error on Pod create could break things like Deployment updates. But if we update defaulting, then I think we won't hit this issue.

#### What would you like to be added:

https://github.com/kubernetes/kubernetes/pull/98866#issuecomment-849203505

**Comment by wzshiming** [comment]
Yeah, we can only do so much more if we don't start rejecting requests.

**Comment by liggitt** [comment]
I think warning and treating as 1 is a reasonable point at which to end this improvement.

**Comment by wzshiming** [comment]
make sense
